BaseLib/ConfigTreeNew.cpp

```diff
@@ -0,0 +1,225 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "ConfigTreeNew.h"
+
+namespace BaseLib
+{
+
+const char ConfigTreeNew::pathseparator = '/';
+const std::string ConfigTreeNew::key_chars_start = "abcdefghijklmnopqrstuvwxyz";
+const std::string ConfigTreeNew::key_chars = key_chars_start + "_0123456789";
+
+ConfigTreeNew::
+ConfigTreeNew(PTree const& tree,
+              Callback const& error_cb,
+              Callback const& warning_cb)
+    : _tree(&tree), _onerror(error_cb), _onwarning(warning_cb)
+{
+    if (!_onerror) {
+        ERR("ConfigTree: No valid error handler provided.");
+        std::abort();
```

Comment:
If the check fails, the program __will__ segfault later on, so I think there is no additional problem with `std::abort()`.
Also, I think calling `std::abort()` in the constructor is as fatal as calling it outside: The program will terminate without cleanup anyway.
Maybe we could discuss, if it is a good idea to use abort() anywhere, because it does not clean up. And maybe in the future it would be nice to clean up, e.g., finish writing to some file or remove temporary files. But that discussion can take place another day.